### ISSUE TITLE
Jumps at intersections to back-and-forth on east-to-west segments.

### ISSUE THREAD

**Issue by FewKinG** [eclipse/sprotty#277]
I came across a bug in the implementation of the `JumpingPolylineView` which causes intersecting edge paths to jump across one another. Everything's working as intended normally but in my diagram I noticed some lines that seemed strange to me:

![image](https://user-images.githubusercontent.com/152299/157264827-84db057e-58bd-4190-8d79-279c48a9dd7d.png)

I noticed that this behavior affected lines going from east to west (the above one in the screenshot). This has to do with how the path is constructed in the `intersectionPath()` method. Intersections are sorted `BY_X_THEN_Y` and if multiple intersections are on that same segment, the end of one intersection connects to the start of the next intersection which causes the line to jump back-and-forth and draw over itself.

To fix this, I extended the `JumpingPolylineView` class, overriding the `intersectionPath` method, to process the intersections in reverse order when the current segment is going the other way:

```    
protected intersectionPath(edge: SEdge, segments: Point[], intersectingPoint: IntersectingRoutedPoint, args?: IViewArgs): string {
        let path = '';
        let intersections = intersectingPoint.intersections.sort(BY_X_THEN_Y);
        const pBefore = segments[segments.indexOf(intersectingPoint) - 1]; // will always be present as the first segment is the starting point
        if (intersectingPoint.x < pBefore.x) {
            // if the segment goes from right to left, we have to add the intersections in different order
            intersections = intersections.reverse();
        }
        for (const intersection of intersections) {
            [...]
```

The rest of the code remains untouched. This bug probably also affects `PolylineEdgeViewWithGapsOnIntersections` but as this uses the same implementation of `intersectionPath` in `JumpingPolylineView` it would also be fixed by this change.

@planger I figure you implemented this feature in the first place. I would love to contribute, but my time is somewhat limited and I'm not so much for creating an Eclipse whatever account to be allowed to open a pull request just for these few lines of code. Also at this point I don't have all the insight to evaluate whether this is the only case where this bug triggers and whether my fix might introduce issues in different scenarios that I'm not seeing right now. In my diagram, I'm only using straight orthogonal edges (north-south, east-west) so there might be stuff going on with slanted lines (although I'm not expecting it). Also I don't know if I'd have to write any tests to ensure this performs properly. 

So if you or anyone else would be so kind to get this bugfix into the library code at some point so others can also benefit from it, I would very much appreciate it. Thank you very much!

Side-note: Using this library has been great so far. The ease of use and design principles used here had me rapidly integrate and modify it to my needs and it works beautifully (I am using it in conjunction with the awesome ELKjs lib).

**Comment by planger** [comment]
Thank you @FewKinG for raising this bug and even proposing a potential fix!

I can reproduce the bug. The use case where I've been using these edge views are indeed only dealing with edges going east-to-west. Therefore I never observed this effect.

I'm quite busy this week, but I'll look into your potential fix next week.

**Comment by planger** [comment]
Just as an update: I looked into your fix and it certainly improves the situation. Thanks a lot for that!

However, I found a few more cases that don't seem to be handled nicely yet (the [random graph](https://github.com/eclipse/sprotty/tree/master/examples/random-graph) is a nice test case for that). I'll have to look into those issues in more detail. Hope I'll find some time next week or the week after.
